### PR TITLE
detect/byte_jump: Implement "from_end" option

### PR DESCRIPTION
**Pull request by jlucovsky**
This PR adds support for `from_end` to the existing `byte_jump`. Although documented, it was not implemented.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [3626](https://redmine.openinfosecfoundation.org/issues/3626)

Describe changes:
- Implement `byte_jump` `"from_end"` option
- Add unit tests using `from_end`

Companion [Suricata-verify PR #232](https://github.com/OISF/suricata-verify/pull/232)

### CODE REVIEW

File: src/detect-bytejump.h [hunk 1]
```diff
@@ -39,11 +39,12 @@
 #define DETECT_BYTEJUMP_ALIGN    0x20 /**< "align" offset */
 #define DETECT_BYTEJUMP_DCE      0x40 /**< "dce" enabled */
 #define DETECT_BYTEJUMP_OFFSET_BE 0x80 /**< "byte extract" enabled */
+#define DETECT_BYTEJUMP_END      0x100 /**< "from_end" jump */
```
Comment (victorjulien) [inline src/detect-bytejump.h:42]: can you convert this to use `BIT_U16()`

File: src/detect-bytejump.c [hunk 2]
```diff
@@ -168,8 +168,10 @@ int DetectBytejumpDoMatch(DetectEngineThreadCtx *det_ctx, const Signature *s,
     if (flags & DETECT_BYTEJUMP_BEGIN) {
         jumpptr = payload + val;
         //printf("NEWVAL: payload %p + %ld = %p\n", p->payload, val, jumpptr);
-    }
-    else {
+    } else if (flags & DETECT_BYTEJUMP_END) {
+        jumpptr = payload + payload_len + val;
+        //printf("NEWVAL: payload %p + %ld = %p\n", p->payload, val, jumpptr);
```
Comment (victorjulien) [inline src/detect-bytejump.c:173]: nit: would be better as `SCLogDebug` instead. Maybe clean the older ones up as well?

File: src/detect-bytejump.c [hunk 3]
```diff
@@ -1094,6 +1099,19 @@ static int DetectBytejumpTestParse12(void)
     return result;
 }
 
+static int DetectBytejumpTestParse13(void)
+{
+    DetectBytejumpData *data = NULL;
+    data=DetectBytejumpParse(NULL, " 4,0 , relative , little, string, dec, "
```
Comment (victorjulien) [inline src/detect-bytejump.c:1105]: style: spaces around `=`. Also, `data` can be declared on this line too.

File: src/detect-bytejump.c [hunk 4]
```diff
@@ -1283,6 +1301,26 @@ static int DetectByteJumpTestPacket07 (void)
...
+ */
+static int DetectByteJumpTestPacket08 (void)
+{
+    uint8_t *buf = (uint8_t *)"XX04abcdABCD";
+    uint16_t buflen = strlen((char *)buf);
+    Packet *p = UTHBuildPacket((uint8_t *)buf, buflen, IPPROTO_TCP);
+
+    FAIL_IF_NULL(p);
+
+    char sig[] = "alert tcp any any -> any any (content:\"XX\"; byte_jump:2,0,relative,string,dec,from_end, post_offset -8; content:\"ABCD\";  sid:1; rev:1;)";
```
Comment (victorjulien) [inline src/detect-bytejump.c:1315]: can you break this line to not exceed ~80 chars?

**Comment by jlucovsky** [comment]
Continued in #4948

**Pull request closed without merging**